sources/Outputs/src/DydVRRemote.cpp

```diff
@@ -23,29 +23,26 @@ namespace outputs {
 void DydVRRemote::writeVRRemotes(boost::shared_ptr<dynamicdata::DynamicModelsCollection> &dynamicModelsToConnect, const std::string &basename) {
   using BusId = std::string;
   std::set<BusId> regulatedBusIds;
-  for (const auto &busToGenerator : busesRegulatedBySeveralGenerators_)
-    regulatedBusIds.insert(busToGenerator.first);
-  for (const auto &busToVSC : hvdcDefinitions_.vscBusVSCDefinitionsMap)
-    regulatedBusIds.insert(busToVSC.first);
-
-  for (const BusId &regulatedBusId : regulatedBusIds) {
-    std::string id = constants::modelSignalNQprefix_ + regulatedBusId;
-    boost::shared_ptr<dynamicdata::BlackBoxModel> blackBoxModelVRRemote = helper::buildBlackBox(id, "VRRemote", basename + ".par", id);
-    dynamicModelsToConnect->addModel(blackBoxModelVRRemote);
-    dynamicModelsToConnect->addConnect(id, "vrremote_URegulated", constants::networkModelName, regulatedBusId + "_U_value");
+  for (const auto &busId2Number : busesToNumberOfRegulationMap_) {
+    if (busId2Number.second == dfl::inputs::NetworkManager::NbOfRegulating::MULTIPLES) {
+      regulatedBusIds.insert(busId2Number.first);
+    }
   }
```

Comment:
Maybe there is something I am missing but I do not see the effect of this loop and of the `regulatedBusIds` set on the following lines of the method.

Reply:
solved